### PR TITLE
chore: lint updates

### PR DESCRIPTION
**Pull request by eh-am**
* Make rules more strict
* Add some documentation to explain what the imports mean

### CODE REVIEW

File: public/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+### Guidelines for imports
+
+It may be confusing to see different imports, so let's go over the most common examples:
+
+`@webapp` -> Refers to `pyroscope-oss`, aka `grafana/pyroscope` repo.
+`@pyroscope/webapp` -> Refers to `pyroscope-oss`, aka `grafana/pyroscope` repo.
+`@phlare` -> Refers to code in this repository. Note that this is needed since other
+downstream repositories may use this repository, and they also may want to override specific files.
+
+In the future, once both `grafana/pyroscope` and `grafana/pyroscope` are merged, there will
```
Comment (hi-rustin) [inline public/README.md:24]:
```suggestion
In the future, once both `grafana/pyroscope` and `grafana/phlare` are merged, there will
```

Do you mean this?

File: scripts/webpack/webpack.common.js [hunk 2]
```diff
@@ -65,10 +65,6 @@ module.exports = {
         __dirname,
         '../../public/app/redux/hooks'
       ),
-      '@webapp/redux/reducers/tenant': path.resolve(
```
Comment (hi-rustin) [inline scripts/webpack/webpack.common.js:68]: But why do we have this here?
```js
      '@webapp/redux/store': path.resolve(
        __dirname,
        '../../public/app/redux/store'
      ),
```

**Review by hi-rustin — APPROVED** [review]
rest LGTM

File: public/README.md (hunk 1 above)
Comment (eh-am) [reply public/README.md:24]: whoops!

File: scripts/webpack/webpack.common.js (hunk 2 above)
Comment (eh-am) [reply scripts/webpack/webpack.common.js:68]: Because of this https://github.com/search?q=repo%3Agrafana%2Fpyroscope%20%40webapp%2Fredux%2Fstore&type=code

Comment (hi-rustin) [reply scripts/webpack/webpack.common.js:68]: Got it. Thanks!

**Review by cyriltovena — APPROVED** [review]
LGTM

**Review by cyriltovena — APPROVED** [review]
LGTM

**Pull request merged**